src/components/core/Xirr.tsx

```diff
@@ -0,0 +1,63 @@
+// (C) 2019 GoodData Corporation
+import * as React from "react";
+
+import {
+    ICommonVisualizationProps,
+    ILoadingInjectedProps,
+    visualizationLoadingHOC,
+    commonDefaultProps,
+} from "./base/VisualizationLoadingHOC";
+import { IDataSourceProviderInjectedProps } from "../afm/DataSourceProvider";
+import { BaseVisualization } from "./base/BaseVisualization";
+import { IntlWrapper } from "./base/IntlWrapper";
+import { IntlTranslationsProvider, ITranslationsComponentProps } from "./base/TranslationsProvider";
+import XirrTransformation from "../visualizations/headline/XirrTransformation";
+import { fixEmptyHeaderItems } from "./base/utils/fixEmptyHeaderItems";
+
+export class XirrStateless extends BaseVisualization<
+    ICommonVisualizationProps & ILoadingInjectedProps & IDataSourceProviderInjectedProps,
+    {}
+> {
+    public static defaultProps: Partial<ICommonVisualizationProps> = commonDefaultProps;
+
+    protected renderVisualization(): JSX.Element {
+        const {
+            afterRender,
+            drillableItems,
+            locale,
+            dataSource,
+            resultSpec,
+            execution,
+            onFiredDrillEvent,
+            onDrill,
+            config,
+        } = this.props;
+
+        return (
+            <IntlWrapper locale={locale}>
+                <IntlTranslationsProvider>
+                    {(props: ITranslationsComponentProps) => (
+                        <XirrTransformation
+                            onAfterRender={afterRender}
+                            onFiredDrillEvent={onFiredDrillEvent}
+                            drillableItems={drillableItems}
+                            onDrill={onDrill}
+                            config={config}
+                            executionRequest={{
+                                afm: dataSource.getAfm(),
+                                resultSpec,
+                            }}
+                            executionResponse={execution.executionResponse}
+                            executionResult={fixEmptyHeaderItems(
+                                execution.executionResult,
+                                props.emptyHeaderString,
+                            )}
```

Comment:
Just a note: executionRequest and executionResult create new objects all the time. This is probably same as everywhere in version 7, but in modern React, this would be considered a bad practice which may result in unnecessary rerenders.

Reply:
True, however we do not memoize it anywehere else, and the object is dependant on props, so I'd leave it as is for consistency sake.